src/poetry/core/packages/package.py

```diff
@@ -481,7 +481,8 @@ def without_optional_dependency_groups(self: T) -> T:
         """
         Returns a clone of the package without optional dependency groups.
         """
-        package = self.clone()
+        package = self.clone_shallow()
+        package._dependency_groups = package._dependency_groups.copy()
```

Comment:
a little comment explaining why we are copying the package._depndency_groups wouldbe great